global.bat

```diff
@@ -234,94 +238,99 @@ rem start "" cmd /c echo todo ^| "%~dp0programs\tvcman.exe" "%~dp0output\encode%
+
+:: Timecodes ::
+echo Generating timecodes...
+".\programs\replacetext" "encode.avs" "i444 = false" "i444 = true"
+".\programs\replacetext" "encode.avs" "pass = 0" "pass = 1"
+".\programs\ffmpeg" -hide_banner -v error -stats -i encode.avs -an -f null -
+".\programs\replacetext" "encode.avs" "pass = 1" "pass = 2"
+
 :: Video ::
-".\programs\avs2pipemod" -y4mp encode.avs | ".\programs\x264_x64" --threads auto --sar "%VAR%" --crf 20 --keyint 600 --ref 16 --no-fast-pskip --bframes 16 --b-adapt 2 --direct auto --me tesa --merange 64 --subme 11 --trellis 2 --partitions all --no-dct-decimate --input-range pc --range pc --tcfile-in ".\temp\times.txt" -o ".\temp\video.mkv" --colormatrix smpte170m --output-csp i444 --profile high444 --output-depth 10 --demuxer y4m -
+echo Encoding video...
+".\programs\x264_x64" --threads auto --sar "%VAR%" --crf 20 --keyint 600 --ref 16 --no-fast-pskip --bframes 16 --b-adapt 2 --direct auto --me tesa --merange 64 --subme 11 --trellis 2 --partitions all --no-dct-decimate --input-range pc --range pc --tcfile-in ".\temp\times.txt" -o ".\temp\video.mkv" --colormatrix smpte170m --output-csp i444 --profile high444 --output-depth 10 encode.avs
+
 :: Muxing ::
 ".\programs\mkvmerge" -o ".\output\encode.mkv" --timecodes -1:".\temp\times.txt" ".\temp\video.mkv" ".\temp\audio.opus"
  if "%EncodeChoice%"=="1" goto Defaults
 
 : 512kb
-title Starting up...
-title
-:: Audio ::
-".\programs\avs2pipemod" -wav encode.avs | ".\programs\qaac64" -v 0 --he -q 2 --delay -5187s --threading --no-smart-padding - -o ".\temp\audio.mp4"
+title Processing Compatability
 echo.
 echo -------------------------------
 echo  Encoding Archive 512kb stream
 echo -------------------------------
 echo.
+:: Audio ::
+echo Encoding audio...
+".\programs\ffmpeg" -y -hide_banner -v error -stats -i encode.avs -vn -c:a libfdk_aac -profile:a aac_he_v2 -vbr 2 ".\temp\audio.mp4"
+
 :: Video ::
+echo Encoding video...
 ".\programs\replacetext" "encode.avs" "pass = 2" "pass = 0"
 ".\programs\replacetext" "encode.avs" "i444 = true" "i444 = false"
-".\programs\avs2pipemod" -y4mp encode.avs | ".\programs\x264_x64" --threads auto --crf 20 --keyint 600 --ref 16 --no-fast-pskip --bframes 16 --b-adapt 2 --direct auto --me tesa --merange 64 --subme 11 --trellis 2 --partitions all --no-dct-decimate --range tv --input-range tv --colormatrix smpte170m -o ".\temp\video_512kb.h264" --demuxer y4m -
+".\programs\x264_x64" --threads auto --crf 20 --keyint 600 --ref 16 --no-fast-pskip --bframes 16 --b-adapt 2 --direct auto --me tesa --merange 64 --subme 11 --trellis 2 --partitions all --no-dct-decimate --range tv --input-range tv --colormatrix smpte170m -o ".\temp\video_512kb.h264" encode.avs
+
 :: Muxing ::
-for /f "tokens=2" %%i in ('^""%~dp0programs\avs2pipemod" -info "%~dp0encode.avs" ^|find "fps"^"') do (set fps=%%i)
+for /f "tokens=2 delims==" %%i in ('FINDSTR "r_frame_rate" "%~dp0temp\info.txt"') do (set fps=%%i)

```

Comment:
Will `info.txt` be there if we only run 512kb alone?

Reply:
It sure will, since the `info.txt` file is generated when determining PAR. However you uncovered another problem - `info.txt` doesn't get generated when PAR is set to 1:1.

Reply:
Fixed by adding another call to `ffprobe` for 1:1 aspect ratios so the framerate will be available later.